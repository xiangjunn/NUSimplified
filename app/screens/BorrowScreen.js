import { Container, Content, Footer, FooterTab, Button, Text, Icon, Header, Body, Left, Right, Label } from 'native-base';
import React from 'react';
import { StyleSheet } from 'react-native';
import { useNavigation } from '@react-navigation/native'

function BorrowScreen() {
    const navigation = useNavigation();
    const homeStyle = styles.others;
    const bookingStyle = styles.others;
    const borrowStyle = styles.selected;
    const remindersStyle = styles.others;

    return (
        <Container >
          <Header androidStatusBarColor='#62B1F6' style={{backgroundColor: '#62B1F6'}}>
          <Left>
            <Button transparent onPress={() => navigation.openDrawer()}>
              <Icon name='menu' />
            </Button>
          </Left>
          <Body/>
          <Right/>
            </Header>
          <Content>
            <Label>Borrow - Not implemented yet</Label>
          </Content>
          <Footer style={{backgroundColor: '#62B1F6'}}>
            <FooterTab>
              <Button info onPress={() => navigation.navigate("Home")}>
                <Icon name='home' style={homeStyle} />
                <Text style={homeStyle} >Home</Text>
              </Button>

              <Button info onPress={() => navigation.navigate("Booking")}>
                  <Icon type='FontAwesome5' name='id-card' style={bookingStyle} />
                <Text style={bookingStyle} >Booking</Text>
              </Button>

              <Button info>
              <Icon name="book" style={borrowStyle} />
              <Text style={borrowStyle}>Borrow</Text>
              </Button>

              <Button info onPress={() => navigation.navigate("Reminder")}>
                  <Icon name='alarm' style={remindersStyle} />
                <Text style={remindersStyle}>Reminder</Text>
              </Button>
            </FooterTab>
          </Footer>
        </Container>
    );
}

const styles = StyleSheet.create({
    selected: {
        color: 'black'
    },
    others: {
        color: 'white'
    }
});

export default BorrowScreen;