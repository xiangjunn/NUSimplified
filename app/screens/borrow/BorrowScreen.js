import { Container, Footer, FooterTab, Button, Text, Icon, Header, Body, Left, Right, Label, List, ListItem, Thumbnail, Form } from 'native-base';
import React, { useState, useEffect} from 'react';
import { Image, StyleSheet } from 'react-native';
import { useNavigation } from '@react-navigation/native'
import { LinearGradient } from 'expo-linear-gradient';

function BorrowScreen() {
    const navigation = useNavigation();
    const homeStyle = styles.others;
    const bookingStyle = styles.others;
    const borrowStyle = styles.selected;
    const remindersStyle = styles.others;

    return (
        <Container >
          <Header androidStatusBarColor='#62B1F6' style={{backgroundColor: '#62B1F6'}}>
          <Left>
            <Button transparent onPress={() => navigation.openDrawer()}>
              <Icon name='menu' />
            </Button>
          </Left>
          <Body/>
          <Right/>
            </Header>
          <Form style={{flex: 1, justifyContent: 'center'}}>
          <LinearGradient
        colors={['#ff7e5f', '#feb47b']}
        start={[0, 1]}
        end={[1, 0]}
        style={styles.view}>
          <Button transparent style={styles.button} onPress={() => navigation.navigate('Browse')}>
          <Label  style={[styles.text, {color: 'black'}]}>Browse</Label>
          <Icon type='FontAwesome5' name='search' style={[styles.icon, {color: 'black'}]}/>
          </Button>
        </LinearGradient>
          <LinearGradient
        colors={['#42275a', '#734b6d']}
        start={[0, 1]}
        end={[1, 0]}
        style={styles.view}>
          <Button transparent style={styles.button} onPress={() => navigation.navigate('Loan')}>
          <Label style={[styles.text, {color: 'white'}]}>Book Loans</Label>
          <Icon type='FontAwesome5' name='book' style={[styles.icon, {color: 'white'}]}/>
          </Button>
        </LinearGradient>
          </Form>
          <Footer style={{backgroundColor: '#62B1F6'}}>
            <FooterTab>
              <Button info onPress={() => navigation.navigate("Home")}>
                <Icon name='home' style={homeStyle} />
                <Text style={homeStyle} >Home</Text>
              </Button>

              <Button info onPress={() => navigation.navigate("Booking")}>
                  <Icon type='FontAwesome5' name='id-card' style={bookingStyle} />
                <Text style={bookingStyle} >Booking</Text>
              </Button>

              <Button info>
              <Icon name="book" style={borrowStyle} />
              <Text style={borrowStyle}>Borrow</Text>
              </Button>

              <Button info onPress={() => navigation.navigate("Reminder")}>
                  <Icon name='alarm' style={remindersStyle} />
                <Text style={remindersStyle}>Reminder</Text>
              </Button>
            </FooterTab>
          </Footer>
        </Container>
    );
}

const styles = StyleSheet.create({
    selected: {
        color: 'black'
    },
    others: {
        color: 'white'
    },
    view: {
      width: '80%',
      height: '40%',
      alignSelf: 'center',
      justifyContent: 'center',
      margin: '7%',
      borderRadius: 40
    },
    text: {
      textAlign: 'center',
      fontSize: 40,
      fontWeight: 'bold'
    },
    icon: {
      alignSelf: 'center',
      fontSize: 100,
      marginTop: 20
    },
    button: {
      flex: 1,
      width: '100%',
      flexDirection: 'column',
      alignSelf: 'center',
      justifyContent: 'center',
      borderRadius: 40
    }
});

export default BorrowScreen;