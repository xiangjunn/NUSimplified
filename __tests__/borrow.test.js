import React from 'react';
import { render, cleanup } from '@testing-library/react-native';
import BorrowScreen from '../app/screens/borrow/BorrowScreen';

afterEach(() => {    
  jest.clearAllMocks();
  cleanup();
});

const mockReplace = jest.fn();
jest.mock("@react-navigation/native", () => {
    const actualNav = jest.requireActual("@react-navigation/native");
    return {
      ...actualNav,
      useNavigation: () => ({
        navigate: jest.fn(),
        replace: mockReplace,
        dispatch: jest.fn(),
      }),
    };
});

window.addEventListener = jest.fn();
window.attachEvent= jest.fn();

describe('Borrow Test', () => {
    test('renders correctly', () => {
        const tree = render(<BorrowScreen />).toJSON();
        expect(tree).toMatchSnapshot();
    });  
});