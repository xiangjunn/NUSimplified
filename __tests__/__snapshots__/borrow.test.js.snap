// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Borrow Test renders correctly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#fff",
      "flex": 1,
      "height": 1334,
    }
  }
>
  <View>
    <RCTSafeAreaView
      emulateUnlessSupported={true}
      style={
        Object {
          "backgroundColor": "#62B1F6",
        }
      }
    >
      <View
        androidStatusBarColor="#62B1F6"
        style={
          Array [
            Object {
              "backgroundColor": "#F8F8F8",
              "borderBottomColor": "#a7a6ab",
              "borderBottomWidth": 0.5,
              "elevation": 3,
              "flexDirection": "row",
              "height": 64,
              "justifyContent": "center",
              "left": 0,
              "paddingLeft": 6,
              "paddingRight": 10,
              "right": 0,
              "shadowColor": undefined,
              "shadowOffset": undefined,
              "shadowOpacity": undefined,
              "shadowRadius": undefined,
              "top": 0,
            },
            Object {
              "backgroundColor": "#62B1F6",
            },
          ]
        }
      >
        <View
          style={
            Object {
              "alignItems": "flex-start",
              "alignSelf": "center",
              "flex": 1,
            }
          }
        >
          <View
            accessible={true}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "opacity": 1,
              }
            }
          >
            <Text />
          </View>
        </View>
        <View
          style={
            Object {
              "alignItems": "center",
              "alignSelf": "center",
              "flex": 1,
            }
          }
        />
        <View
          style={
            Object {
              "alignItems": "flex-end",
              "alignSelf": "center",
              "flex": 1,
              "flexDirection": "row",
              "justifyContent": "flex-end",
            }
          }
        />
      </View>
    </RCTSafeAreaView>
  </View>
  <View
    style={
      Array [
        Object {},
        Object {
          "flex": 1,
          "justifyContent": "center",
        },
      ]
    }
  >
    <ViewManagerAdapter_ExpoLinearGradient
      proxiedProperties={
        Object {
          "colors": Array [
            4294934111,
            4294882427,
          ],
          "endPoint": Array [
            1,
            0,
          ],
          "locations": undefined,
          "startPoint": Array [
            0,
            1,
          ],
        }
      }
      style={
        Object {
          "alignSelf": "center",
          "borderRadius": 40,
          "height": "40%",
          "justifyContent": "center",
          "margin": "7%",
          "width": "80%",
        }
      }
    >
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignSelf": "center",
            "borderRadius": 40,
            "flex": 1,
            "flexDirection": "column",
            "justifyContent": "center",
            "opacity": 1,
            "width": "100%",
          }
        }
      >
        <Text
          style={
            Array [
              Object {
                "fontSize": 17,
              },
              Object {
                "fontSize": 40,
                "fontWeight": "bold",
                "textAlign": "center",
              },
              Object {
                "color": "black",
              },
            ]
          }
        >
          Browse
        </Text>
        <Text />
      </View>
    </ViewManagerAdapter_ExpoLinearGradient>
    <ViewManagerAdapter_ExpoLinearGradient
      proxiedProperties={
        Object {
          "colors": Array [
            4282525530,
            4285746029,
          ],
          "endPoint": Array [
            1,
            0,
          ],
          "locations": undefined,
          "startPoint": Array [
            0,
            1,
          ],
        }
      }
      style={
        Object {
          "alignSelf": "center",
          "borderRadius": 40,
          "height": "40%",
          "justifyContent": "center",
          "margin": "7%",
          "width": "80%",
        }
      }
    >
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignSelf": "center",
            "borderRadius": 40,
            "flex": 1,
            "flexDirection": "column",
            "justifyContent": "center",
            "opacity": 1,
            "width": "100%",
          }
        }
      >
        <Text
          style={
            Array [
              Object {
                "fontSize": 17,
              },
              Object {
                "fontSize": 40,
                "fontWeight": "bold",
                "textAlign": "center",
              },
              Object {
                "color": "white",
              },
            ]
          }
        >
          Book Loans
        </Text>
        <Text />
      </View>
    </ViewManagerAdapter_ExpoLinearGradient>
  </View>
  <RCTSafeAreaView
    emulateUnlessSupported={true}
    style={
      Object {
        "backgroundColor": "#62B1F6",
      }
    }
  >
    <View
      style={
        Array [
          Object {
            "backgroundColor": "#F8F8F8",
            "borderColor": "#cbcbcb",
            "borderTopWidth": 0.5,
            "elevation": 3,
            "flexDirection": "row",
            "height": 55,
            "justifyContent": "center",
            "left": 0,
            "paddingBottom": 0,
            "right": 0,
          },
          Object {
            "backgroundColor": "#62B1F6",
          },
        ]
      }
    >
      <View
        style={
          Object {
            "alignSelf": "stretch",
            "backgroundColor": undefined,
            "flex": 1,
            "flexDirection": "row",
            "justifyContent": "space-between",
          }
        }
      >
        <View
          accessible={true}
          focusable={true}
          forwardedRef={[Function]}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "alignSelf": "center",
              "backgroundColor": "#62B1F6",
              "borderBottomWidth": null,
              "borderColor": null,
              "borderLeftWidth": null,
              "borderRadius": 5,
              "borderRightWidth": null,
              "borderTopWidth": null,
              "borderWidth": undefined,
              "elevation": 0,
              "flex": 1,
              "flexDirection": null,
              "height": 55,
              "justifyContent": "center",
              "opacity": 1,
              "paddingBottom": 6,
              "paddingTop": 6,
              "shadowColor": null,
              "shadowOffset": null,
              "shadowOpacity": null,
              "shadowRadius": null,
            }
          }
        >
          <Text />
          <Text
            style={
              Array [
                Object {
                  "backgroundColor": "transparent",
                  "color": "#6b6b6b",
                  "fontFamily": "System",
                  "fontSize": 14,
                  "lineHeight": 16,
                  "marginLeft": 0,
                  "marginRight": 0,
                  "paddingLeft": 16,
                  "paddingRight": 16,
                },
                Object {
                  "color": "white",
                },
              ]
            }
            uppercase={false}
          >
            Home
          </Text>
        </View>
        <View
          accessible={true}
          focusable={true}
          forwardedRef={[Function]}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "alignSelf": "center",
              "backgroundColor": "#62B1F6",
              "borderBottomWidth": null,
              "borderColor": null,
              "borderLeftWidth": null,
              "borderRadius": 5,
              "borderRightWidth": null,
              "borderTopWidth": null,
              "borderWidth": undefined,
              "elevation": 0,
              "flex": 1,
              "flexDirection": null,
              "height": 55,
              "justifyContent": "center",
              "opacity": 1,
              "paddingBottom": 6,
              "paddingTop": 6,
              "shadowColor": null,
              "shadowOffset": null,
              "shadowOpacity": null,
              "shadowRadius": null,
            }
          }
        >
          <Text />
          <Text
            style={
              Array [
                Object {
                  "backgroundColor": "transparent",
                  "color": "#6b6b6b",
                  "fontFamily": "System",
                  "fontSize": 14,
                  "lineHeight": 16,
                  "marginLeft": 0,
                  "marginRight": 0,
                  "paddingLeft": 16,
                  "paddingRight": 16,
                },
                Object {
                  "color": "white",
                },
              ]
            }
            uppercase={false}
          >
            Booking
          </Text>
        </View>
        <View
          accessible={true}
          focusable={false}
          forwardedRef={[Function]}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "alignSelf": "center",
              "backgroundColor": "#62B1F6",
              "borderBottomWidth": null,
              "borderColor": null,
              "borderLeftWidth": null,
              "borderRadius": 5,
              "borderRightWidth": null,
              "borderTopWidth": null,
              "borderWidth": undefined,
              "elevation": 0,
              "flex": 1,
              "flexDirection": null,
              "height": 55,
              "justifyContent": "center",
              "opacity": 1,
              "paddingBottom": 6,
              "paddingTop": 6,
              "shadowColor": null,
              "shadowOffset": null,
              "shadowOpacity": null,
              "shadowRadius": null,
            }
          }
        >
          <Text />
          <Text
            style={
              Array [
                Object {
                  "backgroundColor": "transparent",
                  "color": "#6b6b6b",
                  "fontFamily": "System",
                  "fontSize": 14,
                  "lineHeight": 16,
                  "marginLeft": 0,
                  "marginRight": 0,
                  "paddingLeft": 16,
                  "paddingRight": 16,
                },
                Object {
                  "color": "black",
                },
              ]
            }
            uppercase={false}
          >
            Borrow
          </Text>
        </View>
        <View
          accessible={true}
          focusable={true}
          forwardedRef={[Function]}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "alignSelf": "center",
              "backgroundColor": "#62B1F6",
              "borderBottomWidth": null,
              "borderColor": null,
              "borderLeftWidth": null,
              "borderRadius": 5,
              "borderRightWidth": null,
              "borderTopWidth": null,
              "borderWidth": undefined,
              "elevation": 0,
              "flex": 1,
              "flexDirection": null,
              "height": 55,
              "justifyContent": "center",
              "opacity": 1,
              "paddingBottom": 6,
              "paddingTop": 6,
              "shadowColor": null,
              "shadowOffset": null,
              "shadowOpacity": null,
              "shadowRadius": null,
            }
          }
        >
          <Text />
          <Text
            style={
              Array [
                Object {
                  "backgroundColor": "transparent",
                  "color": "#6b6b6b",
                  "fontFamily": "System",
                  "fontSize": 14,
                  "lineHeight": 16,
                  "marginLeft": 0,
                  "marginRight": 0,
                  "paddingLeft": 16,
                  "paddingRight": 16,
                },
                Object {
                  "color": "white",
                },
              ]
            }
            uppercase={false}
          >
            Reminder
          </Text>
        </View>
      </View>
    </View>
  </RCTSafeAreaView>
</View>
`;
